internal/manifest/list.go

```diff
@@ -88,11 +89,12 @@ type ListEdit struct {
 	UpdateMediaType string
 
 	// If Op = ListEditAdd. All fields must be set.
-	AddDigest      digest.Digest
-	AddSize        int64
-	AddMediaType   string
-	AddPlatform    *imgspecv1.Platform
-	AddAnnotations map[string]string
+	AddDigest                digest.Digest
+	AddSize                  int64
+	AddMediaType             string
+	AddPlatform              *imgspecv1.Platform
+	AddAnnotations           map[string]string
+	AddCompressionAnnotation []compression.Algorithm
```

Comment:
`AddCompressionAlgorithms[Used]`? That this is stored in an annotation is conceptually manifest-format-specific (some future format might have this as a native field).